presto-product-tests/src/main/java/com/facebook/presto/tests/TestGroups.java

```diff
@@ -42,6 +42,11 @@
     public static final String STRING_FUNCTIONS = "string_functions";
     public static final String MATH_FUNCTIONS = "math_functions";
     public static final String STORAGE_FORMATS = "storage_formats";
+    public static final String PROFILE_SPECIFIC_TESTS = "profile_specific_tests";
+    public static final String SINGLENODE = "singlenode";
```

Comment:
I'm just wondering whether it is a good idea to strictly bound test groups to the environments we have pre-defined by the `docker-compose` files. We have pre-defined environments to let the user run product tests easily. But it doesn't mean that those environments are the only environments where those tests could possibly be run. 

Moreover, we are going to add predefined environments for different version of `HDP` and different versions of `CDH`. So instead of having single `singlenode_hdfs_impersonation` environment, we might potentially have  multiple environments, say `cdh5_singlenode_hdfs_impersonation`, `cdh4_singlenode_hdfs_impersonation`, `hdp100500_singlenode_hdfs_impersonation`, etc. And we still need to run impersonation product tests on all of them. 

I would rather leave `profile_specific_tests` group as is. This group does make sense to me, cause it tells that this group of tests shouldn't be widely run on all the environments. But instead of having group per environment i would rather have just 2 groups: `hdfs_impersonation` and `hdfs_no_impersonation`.